maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/MavenDownloadsTestCase.java

```diff
@@ -56,6 +65,60 @@ public void testDownloadFromGivenDestinationWithoutUsingCacheAndExtract() throws
         verifyExtraction(2, "resolver-3.0.0-alpha-1", "resolver-3.0.0-alpha-3");
     }
 
+    @Test
+    public void testDownloadInMultipleThreads() throws IOException, InterruptedException {
+        // cleanup
+        FileUtils.deleteDirectory(targetMavenDir);
+
+        // multiple download
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
well, this wouldn't do the job - I need to start both threads at once - at the very same time.
In addition - your code would freeze - you are waiting for a latch that will be counted down in a future by the same code...